src/Installer.php

```diff
@@ -38,31 +38,38 @@
     private $factory;
 
     /**
+     * @var IOInterface
+     */
+    private $io;
+
+    /**
      * @param ManagerState $state
      * @param Downloader $downloader
      * @param Filesystem $filesystem
      * @param Factory $factory
+     * @param IOInterface $io
      */
     public function __construct(
         ManagerState $state,
         Downloader $downloader,
         Filesystem $filesystem,
-        Factory $factory
+        Factory $factory,
+        IOInterface $io
     ) {
         $this->state      = $state;
         $this->downloader = $downloader;
         $this->filesystem = $filesystem;
         $this->factory    = $factory;
+        $this->io         = $io;
     }
 
     /**
      * @param Workshop $workshop
-     * @param IOInterface $io
      * @throws \League\Flysystem\FileExistsException
      */
-    public function installWorkshop(Workshop $workshop, IOInterface $io)
+    public function installWorkshop(Workshop $workshop)
     {
-        if ($this->state->isWorkshopInstalled($workshop->getName())) {
+        if ($this->state->isWorkshopInstalled($workshop)) {
             throw new WorkshopAlreadyInstalledException;
```

Comment:
ahh missed this... will add asap makes sense rather than relying on the caller/exception handler to already have the workshop